src/core/main.c

```diff
@@ -355,11 +368,9 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         } else if (streq(key, "systemd.confirm_spawn") && value) {
 
-                r = parse_boolean(value);
+                r = parse_confirm_spawn(value, &arg_confirm_spawn);
```

Comment:
uh oh. I am not sure I like this. All our parse_xyz() functions return the parsed data in the last argument, and this means they really only *return* it there. The parameter is never read, it's only overriden. And I am pretty sure we should do the same here: parse_confirm_spawn() should not look into the parameter at all, it should only override it.

(Alternative: choose a different name for this, maybe "update_confirm_spawn()" that makes clear that this function is not of this kind, and actually updates the existing value, and doesn't blindly override it)